### ISSUE TITLE
When does the code become public?

### ISSUE THREAD

**Issue by newjavaer** [ZHKKKe/MODNet#4]

**Comment by ZHKKKe** [comment]
Hi, Thanks for your Attention!

Based on current progress, the online video matting demo is expected to be released after two weeks (including inference code and pre-trained model). The delay of the demo release is due to the need to collect new video data for generalizing the model to the domain of WebCam, so that users can try it online through Google Colab. Besides, the training code will be released after the demo.

**Comment by thiagomcoutinho** [comment]
Hi, first of all, great work!
It's possible to release the pre-trained model on the paper dataset? I'm looking for testing the model on some images that doesn't became from a WebCam. 

If you could release a colab code with this model I think some of the requests will be satisfied.

**Comment by ZHKKKe** [comment]
@thiagomcoutinho 
Oh, yes. We plan to release an image matting demo that contains the model pre-trained on the paper dataset, together with the aforementioned video matting demo after two weeks. Therefore, you can use the image demo to validate other images.

**Comment by ak9250** [comment]
@ZHKKKe thanks, could you also add using a input video instead of a webcam feed for the video colab demo?

**Comment by ZHKKKe** [comment]
@ak9250 
Yep. I will do it in next week.

**Comment by newjavaer** [comment]
@ZHKKKe  Hi!Thank you for your attention about my question.Now,I have another quesiton for you.Is the function of the output of branch D to update the parameters of branch D by comparing with groundtruth?And the layer before the DETAILS DP is a convolutional layer with an activation function?Can I understand this way?

**Comment by ZHKKKe** [comment]
@newjavaer 
For your questions:

**Q1: Is the function of the output of branch D to update the parameters of branch D by comparing with groundtruth?**
When calculating the loss of d_p, we need to use a mask, named m_d in the paper, to apply the loss in the portrait boundaries. Please refer to Eq.3 in our paper for more details.

**Q2: the layer before the DETAILS DP is a convolutional layer with an activation function?**
No. The layer before d_p is a convolutional layer followed by a sigmoid function, which is the same as the last layer of the fusion branch.

**Comment by newjavaer** [comment]
@ZHKKKe Thank you！

**Comment by 2120160608** [comment]
I've tried your OFD method. White edges will appear when the characters move quickly. Do you encounter this problem

**Comment by ZHKKKe** [comment]
@2120160608 
Hi.
As discussed in our paper, the OFD trick only suitable for smooth movements. For reference, I think that the movements in all GIF demos (shown in the README file) can support OFD.

**Comment by 2120160608** [comment]
When will the training code be released? Thank you

**Comment by ZHKKKe** [comment]
@2120160608 
A shown in the "TO DO" section of the README file. The training code will be released next month.

**Comment by 2120160608** [comment]
> @2120160608
> A shown in the "TO DO" section of the README file. The training code will be released next month.
Thank you for your attention and reply. I admire your project very much. Do you have any other methods to solve the edge jitter problem of video segmentation? My wechat 601663546， I am a student of Beijing Institute of technology. I hope to be friends with you.

**Comment by 2120160608** [comment]
Thank you for your attention and reply. I admire your project very much. Do you have any other methods to solve the edge jitter problem of video segmentation? My wechat 601663546， I am a student of Beijing Institute of technology. I hope to be friends with you.

**Comment by ZHKKKe** [comment]
@2120160608 Sure. I will add you :) Thanks

**Comment by ZHKKKe** [comment]
@ak9250 
Hi, our demo for custom videos is now available. Sorry for the delay.
